indexing-service/src/main/java/org/apache/druid/indexing/common/actions/UpdateStatusAction.java

```diff
@@ -34,13 +34,24 @@ public class UpdateStatusAction implements TaskAction<Void>
 {
   @JsonIgnore
   private final String status;
+  @JsonIgnore
+  private final TaskStatus statusFull;
+
+  public UpdateStatusAction(
+      String status
+  )
+  {
+    this(status, null);
+  }
 
   @JsonCreator
   public UpdateStatusAction(
-      @JsonProperty("status") String status
+      @JsonProperty("status") String status,
+      @JsonProperty("statusFull") TaskStatus statusFull
   )
   {
     this.status = status;
+    this.statusFull = statusFull;
```

Comment:
If `status` is deprecated, and callers are expected to use `statusFull` can we add validation in the constructor to make sure only 1 field is set.

Reply:
That's fine. 

My thinking was that in the perform method either status or statusFull is used. So if a caller calls this method with both set, it's likely a mistake, and they won't know about it unless they closely inspect the results of the perform method.